### ISSUE TITLE
[Config] Add Global Labels

### ISSUE THREAD

**Issue by jackshirazi** [elastic/elastic-otel-java#302]
### Classic agent link
- https://www.elastic.co/guide/en/apm/agent/java/current/config-core.html#config-global-labels

### What it does
Labels added to all events, with the format key=value[,key=value[,...]]. Any labels set by application via the API will override global labels with the same keys. The server sees `labels.key=value` attributes

### Implementation difficulty, location, and dynamism
- This can be implemented as an Exporter (it can be added very easily to ElasticSpanExporter). It could also be implemented as a SpanProcessor eg https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/2163 but for performance, especially if it's dynamic, I think it's better in the Exporter
- It can be contributed upstream (but probably not worth it)
- It can be dynamic

**Comment by felixbarny** [comment]
Isn't `OTEL_RESOURCE_ATTRIBUTES` the equivalent for this in the OTel world?

**Comment by jackshirazi** [comment]
Yes, you're right. @akhileshpok please can you feedback on whether this needs to be a dynamically alterable capability, or if the static already existing otel implementation is sufficient

**Comment by felixbarny** [comment]
For reference, `global_labels` isn't a dynamic config option, either.

**Comment by SylvainJuge** [comment]
Another benefit of using otel resource attributes for implementation is that it would allow to provide a cross-platform compatibility, we just have to formally have a specification on the resource attribute key and properly map it into our data model on ingest.
